### ISSUE TITLE
Suppress logging of MQTT password

### ISSUE THREAD

**Issue by rotx** [hjdhjd/homebridge-unifi-protect#104]
**Is your feature request related to a problem? Please describe:**
When configuring an MQTT password, it's logged to the default system log facility.

**Describe the solution you'd like:**
It would be great if the password could be suppressed or "X"-ed out.

**Describe alternatives you've considered:**
N/A

**Additional context:**
Example:

```
Sep 26 21:09:57 automate nvm-exec[3532]: [9/26/2020, 9:09:57 PM] [UniFi Protect]
Protect [UCK-G2-PLUS]: Connected to MQTT broker: mqtt://doorbell:secretpassword@192.168.111.11 (topic: unifi/protect).
```

Suggested solution:
```
Sep 26 21:09:57 automate nvm-exec[3532]: [9/26/2020, 9:09:57 PM] [UniFi Protect]
Protect [UCK-G2-PLUS]: Connected to MQTT broker: mqtt://doorbell:XXXXXXXXXXXXX@192.168.111.11 (topic: unifi/protect).
```

**Comment by hjdhjd** [comment]
Unlikely to implement this. It may make one feel better aesthetically, but it provides exactly zero security whatsoever. If you're accessing the Homebridge logs, you can also access the Homebridge `config.json`.

**Comment by rotx** [comment]
It's true that on the local system I need a higher privilege to view logs than the user running homebridge. However, I'm shipping all logs to a separate log collector, so it's not as clear-cut. Suppressing the password from the log messages might therefore actually be useful.

**Comment by hjdhjd** [comment]
And let's not mention that the URL you posted is insecure to begin with since it's not TLS-encrypted. :smile:

Fair enough...not a use case I envisioned, but still puzzled about why on earth you (or anyone really) would want to ship logs elsewhere in a home setting. Sure, in an enterprise setting...but is this not a wee bit overkill? Tell me what I'm not seeing...because you're right, hadn't considered that use case. :smile:

**Comment by ccutrer** [comment]
/me raises hand as the guy running MQTT over TLS at my house with authentication. Though I don’t (currently) ship my logs. I could see it though, if I had several docker containers for each service, to put them all on the host. I currently have 8 different background daemons running bridging various devices to MQTT.

**Comment by rotx** [comment]
While I tend towards overkill, a log collector is actually useful (somehow my home network has grown to over 100 real and virtual devices over the years, I swear I don't know how that happened). After experimenting a bit, I am using graylog. Especially with Ubiquiti devices it can be useful to correlate their latest firmware upgrade fail. Side note: The MQTT URL is actually secure because it's running on localhost (I just put in a random IP to make the point). And btw thanks for the work on this plugin, I just switched from camera-ffmpeg that I was running for years.

**Comment by hjdhjd** [comment]
Can you all just go away and just appreciate that I implemented MQTT in the first place? :smile:

I've got almost 600 devices on my home network...so I get complexity, and I don't do this because really, it's not necessary unless you have deep-seated OCD. :smile: If you have that much time to look through logs...knock yourself out.

I'll look into dealing with this in a future update, but I make no promises.

Oh...and upgrading to the latest Ubiquiti firmware is an exercise in frustration...virtually nobody should do it until their production firmware releases have been out there for a while and the kinks worked out...and there are almost always kinks.

**Comment by hjdhjd** [comment]
Implemented in v3.7.7.
